x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/function/Function.java

```diff
@@ -42,6 +44,11 @@ public Nullability nullable() {
         return Expressions.nullable(children());
     }
 
+    /** Return a predicate that checks if this function can be used by a provided {@link XPackLicenseState}.*/
+    public Predicate<XPackLicenseState> getLicenseChecker() {
+        return license -> true;
```

Comment:
Because the license is checked via a LicenseFeature, which it has two flavours,  LicenseFeature.Persistent and LicenseFeature.Momentary. I must say I don't really know the difference between them but in order to check the license we have three options:

1.- LicenseFeature#checkWithoutTracking -> This is part of the LicenseFeature base class, and therefore available for both implementations
2.-  LicenseFeature.Momentary#check -> Only for the Momentary flavour.
3.- LicenseFeature.Persistent#checkAndStartTracking -> Only for the persistent.

Not sure which one we should use or we should be able to use any of them. The predicate enables the latter.

Reply:
Maybe a boolean checkLicense(XPackLicenseState state)  makes more sense.

Reply:
I changed it, it is simpler :)